### PR TITLE
Adding insecure server boolean to chart.

### PR DESCRIPTION
**Pull request by bjoernw**
The chart was lacking support for a way to run pomerium in insecure mode. My use case is that I would like to fold pomerium into my istio mesh and since istio already handles mutual-tls I did not need/want pomerium to do that as well. Since pomerium itself already supports an insecure mode I just needed to expose this toggle in the chart.

However, just setting environment variables INSECURE_SERVER and GRPC_INSECURE wasn't enough to get past the failing liveness and readiness checks which still had a scheme of "HTTPS" regardless of the value of INSECURE_SERVER or GRPC_INSECURE.

Here I create a single flag `config.insecure` that abstracts those details away. 

Let me know what you think.

### CODE REVIEW

File: charts/pomerium/templates/NOTES.txt [hunk 1]
```diff
@@ -48,3 +48,11 @@ the official documentation:
 
     https://www.pomerium.io/docs/identity-providers.html
 {{- end }}
+
+{{- if .Values.config.insecure }}
+##############################################################################
+####        DANGER: You have disabled TLS!
+####
+####        Please only do this if you absolutely know what you are doing.
+##############################################################################
```
Comment (desimone) [inline charts/pomerium/templates/NOTES.txt:57]: 👍 Love the disclaimer.

**Comment by desimone** [comment]
@bjoernw just needs a version bump

**Comment by bjoernw** [comment]
@desimone What version do you want me to bump this to? 8.3.0? Should I capture my change in the changelog in the readme?

**Comment by desimone** [comment]
>@desimone What version do you want me to bump this to? 8.3.0? 

Seems good to me, going to ping @travisgroth to make sure this is a minor not a major version bump. I don't think it breaks existing so... should be fine. 

> Should I capture my change in the changelog in the readme?

Please do!

**Comment by bjoernw** [comment]
@desimone bumped it to 8.4.0 and added changelog. Looks good?

File: charts/pomerium/README.md [hunk 2]
```diff
@@ -240,6 +242,10 @@ A full listing of Pomerium's configuration variables can be found on the [config
 
 ## Changelog
 
+### 8.0.4
```
Comment (desimone) [inline charts/pomerium/README.md:245]: typo I think... `8.4.0` right?

Comment (bjoernw) [reply charts/pomerium/README.md]: oops, my bad, fixed

**Comment by desimone** [comment]
Thanks @bjoernw !

**Pull request merged**